### PR TITLE
MAINTAINERS: point "Drivers: Audio" and "Intel (Xtensa)" at each other

### PR DESCRIPTION
**Pull request by marc-hb**
Right now these two sections largely duplicate each other which proves
there is a good chance that someone adding themselves to one will likely
want to add themselves to the other section too.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Review by aborisovich — APPROVED** [review]

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -483,6 +483,7 @@ Documentation:
     labels:
         - "area: ADC"
 
+# See also "Intel Platforms (Xtensa)"
```
Comment (nashif) [inline MAINTAINERS.yml:486]: There is not relation here, there are audio drivers in drivers/audio that are not related to intel platforms, it just happens that the first drivers in audio were for an Intel platform.

**Review by nashif — CHANGES_REQUESTED** [review]
I do not this comment provides any value and is misleading, we do need to however change maintainers an reconsider the area as a whole tbh (Drivers: Audio)
